client/benchmark/src/main/java/org/elasticsearch/client/benchmark/ops/bulk/BulkBenchmarkTask.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.benchmark.ops.bulk;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.client.benchmark.BenchmarkTask;
+import org.elasticsearch.client.benchmark.metrics.Sample;
+import org.elasticsearch.client.benchmark.metrics.SampleRecorder;
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.common.io.PathUtils;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.ESLoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class BulkBenchmarkTask implements BenchmarkTask {
+    private final BulkRequestExecutor requestExecutor;
+    private final String indexFilePath;
+    private final int totalIterations;
+    private final int bulkSize;
+    private LoadGenerator generator;
+    private ExecutorService executorService;
+
+    public BulkBenchmarkTask(BulkRequestExecutor requestExecutor, String indexFilePath, int totalIterations, int bulkSize) {
+        this.requestExecutor = requestExecutor;
+        this.indexFilePath = indexFilePath;
+        this.totalIterations = totalIterations;
+        this.bulkSize = bulkSize;
+    }
+
+    @Override
+    @SuppressForbidden(reason = "PathUtils#get is fine - we don't have environment here")
+    public void setUp(SampleRecorder sampleRecorder) {
+        BlockingQueue<List<String>> bulkQueue = new ArrayBlockingQueue<>(256);
+
+        BulkIndexer runner = new BulkIndexer(bulkQueue, totalIterations, sampleRecorder, requestExecutor);
+
+        executorService = Executors.newSingleThreadExecutor((r) -> new Thread(r, "bulk-index-runner"));
+        executorService.submit(runner);
```

Comment:
I think the more robust approach really is to shutdown the pool and wait for termination with a timeout because this will also work without changes should we change the number of request threads. The separate thread is just needed to decouple file reading from sending requests and I wouldn't expect a large delay there (that's why I set it only to 1 minute before and I didn't run into the `shutdownNow()` case either). The timeout is just there to avoid hanging endlessly for now reason in case something goes wrong very badly.

I'd use `Future.get()` only if I were interested in the result (i.e. the return value of `Future.get()`) but not as a synchronizer in this case. Wdyt?